contrib/mesos/pkg/queue/delay.go

```diff
@@ -353,8 +378,7 @@ func (q *DelayFIFO) pop(cancel <-chan struct{}) interface{} {
 					// we may not have the lock yet, so
 					// broadcast to abort Wait, then
 					// return after lock re-acquisition
-					q.cond().Broadcast()
-					<-signal
+					sitAndSpin(q.cond(), signal)
```

Comment:
if a `cancel` event is processed and fires a broadcast before the above goroutine invokes `Wait()` then receiving from `signal` could block until the next change to the fifo. this change attempts to re-broadcast until `signal` closes